generate_diagrams.py

```
@@ -0,0 +1,200 @@
+    
+    if result.returncode != 0:
+        print(f"Warning: Failed to generate package diagram: {result.stderr.decode()}")
+    else:
+        print(f"UML diagrams saved to {output_dir}")
+
+
+def extract_imports(file_path):
+    """Extract imports from a Python file"""
+    imports = []
+    try:
+        with open(file_path, 'r', encoding='utf-8') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)